src/aqua/decider/deal_storage.aqua

```diff
@@ -21,18 +21,18 @@ func store_deal_state(decider_id: SpellId, deal_id: DealId, state: DealState):
     json <- Json.stringify(state)
     Spell.set_string(deal_id, json)
 
-func get_deal_state(decider_id: SpellId, deal_id: DealId) -> ?DealState:
+func get_deal_state(decider_id: SpellId, deal_id: string, key: string) -> ?DealState:
```

Comment:
```suggestion
func get_deal_state(decider_id: SpellId, deal_id: string, storage_key: string) -> ?DealState:
```
maybe like that a little bit better? not sure

Overall it seems like might make sense to separate `parse_deal_state` function from `load_removed_deal_state` & `load_deal_state`, then it might be a tad easier

but that's not critical